master/internal/db/postgres.go

```diff
@@ -1813,24 +1813,42 @@ func (db *PgDB) Query(queryName string, v interface{}, args ...interface{}) erro
 		vValue := reflect.ValueOf(v).Elem()
 		vValue.Set(reflect.MakeSlice(vValue.Type(), 0, 0))
 		for rows.Next() {
-			sValue := reflect.New(vType.Elem())
-			if serr := rows.StructScan(sValue.Interface()); serr != nil {
-				return serr
+			switch k := vValue.Type().Elem().Kind(); k {
+			case reflect.Ptr:
+				sValue := reflect.New(vValue.Type().Elem().Elem())
+				if err = p(rows, sValue.Interface()); err != nil {
+					return err
+				}
+				vValue = reflect.Append(vValue, sValue)
+			case reflect.Struct:
+				sValue := reflect.New(vValue.Type().Elem())
+				if err = p(rows, sValue.Interface()); err != nil {
+					return err
+				}
+				vValue = reflect.Append(vValue, sValue.Elem())
+			default:
+				return errors.Errorf("unexpected type: %s", k)
 			}
-			vValue = reflect.Append(vValue, sValue.Elem())
 		}
 		reflect.ValueOf(v).Elem().Set(vValue)
 		return nil
 	case reflect.Struct:
 		if rows.Next() {
-			return rows.StructScan(v)
+			return p(rows, v)
 		}
 		return ErrNotFound
 	default:
 		panic(fmt.Sprintf("unsupported query type: %s", kind))
 	}
 }
 
+// Query returns the result of the query. Any placeholder parameters are replaced
+// with supplied args.
```

Comment:
non-blocking: add comments to this function and query proto to make the distinction that Query expects structs with the `db` struct tag and Query proto just needs a protobuf message.